### DISCUSSION TITLE
Create a new command which cleans ionide cache + obj and bin

### DISCUSSION THREAD

**Discussion by forki** [ionide/ionide-vscode-fsharp#1360] [Ideas]
We should have a command that properly cleans bin and obj. Because reasons

**Comment by jbeeko** [comment]
git clean -xfd or -Xfd. The first removes all files that are git ignored as well as files not yet added. But it does not remove the current changes. The second retains new files not yet added to repo.  While an alias to one of these, I would vote for -Xfd, may be useful I see no reason to write anything custom.

**Comment by MangelMaxime** [comment]
The problem with these commands @jbeeko is that they remove everything independently of if they are the obj or bin folder.

For example, if you install your dependencies locally inside your workspace they will also be removed and you will have to re-download/re-install them.

This is often the case when working on a web application like when using Fable or even raw JavaScript because `npm`, `yarn`, etc. install the dependencies in the workspace.

It is also important to note that they are `git` commands and perhaps not everyone is using `git` as their versioning system.

**Comment by jbeeko** [comment]
@MangelMaxime True they will remove everything in .gitignore. I think that is a benefit because when things are going wrong I usually want a reboot back to just my work. But I can see others may not want that. I believe git clean will take patterns so it can target just those directories. 

You point on git not being universally present is well taken.

**Comment by Krzysztof-Cieslak** [comment]
I mean, this issue is "on hold" not because we don't know how to delete the files - this is the easy part. But `obj` folder is important from project system point of view, and removing it while IDE is running has huge implications for IDE behaviour, and things like automatic restoring, triggering design-time builds etc.
